core/global/input_manager.gd

```diff
@@ -147,23 +159,42 @@ func _on_gamepad_change(_device: int, _connected: bool) -> void:
 	for gamepad in managed_gamepads.values():
 		if gamepad.phys_path in discovered_paths:
 			continue
-		logger.debug("Gamepad no longer exists: " + gamepad.phys_path)
-		gamepad.virt_device.close()
+
+		logger.debug("Gamepad disconnected: " + gamepad.phys_path)
 		# Lock the gamepad mappings so we can alter them.
 		gamepad_mutex.lock()
+		orphaned_gamepads[gamepad.phys] = gamepad
 		managed_gamepads.erase(gamepad.phys_path)
-		virtual_gamepads.erase(gamepad.virt_path)
 		gamepad_mutex.unlock()
 
 	# Add any newly found gamepads
 	for path in discovered_paths:
+		# Reject managed and virtual devices
 		if path in managed_gamepads:
 			logger.debug("Gamepad is already being managed: " + path)
 			continue
 		if path in virtual_gamepads:
-			logger.debug("Gamepad appears to be virtual: " + path)
+			logger.debug("Virtual gamepad is already being managed: " + path)
+			continue
+		var input_device := InputDevice.new()
+		if input_device.open(path) != OK:
+			logger.warn("Unable to create managed gamepad for: " + path)
+			continue
+		if input_device.get_phys() == "":
+			logger.debug("Device appears to be virtual, skipping " + path)
+			continue
+		if "Receiver" in input_device.get_name():
+			logger.debug("Device is a receiver, skipping " + path)
+			continue
```

Comment:
We should use a configurable blacklist for devices we don't want to manage.

Reply:
Agreed. I'm not sure how the config is set up to add this though.

Reply:
You can use `SettingsManager` with a default. But for now we could probably add a `# TODO` on there.